### PR TITLE
Wall and girder construction/deconstruction

### PR DESCRIPTION
**Pull request by ike709**
You can now fully construct/deconstruct girders and walls. Girder displacement is also in. Have fun blocking off the spawn area with walls.

I'm not sure if sounds are working for it or not because I didn't bother looking for my headphones when I tested. And I'm not sure if welders use fuel when deconning walls, since I haven't looked at welder code.

Pic:
image

False walls and reinforced walls coming... at some point.

### CODE REVIEW

**Comment by ike709** [comment]
Also I didn't bother with TG's `explosion_block`, `thermal_conductivity`, or `heat_capacity` vars since I'm not sure if those are implemented here or not.

File: code/game/objects/items/stacks/sheets/glass.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ RGlassSheet.template = {
 				merge_type: "RGlassSheet",
 				novariants: false,
 				recipes: [
-					{name: "fulltile reinforced window", template_name: "r_window_construct", cost: 2, time: 200, cant_cross: ["Window"], on_floor: true}
+					{name: "fulltile reinforced window", template_name: "r_window_construct", cost: 2, time: 2000, cant_cross: ["Window"], on_floor: true}
```
Comment (ike709) [inline code/game/objects/items/stacks/sheets/glass.js]: I assumed this was a typo so I fixed it.

File: code/game/objects/items/stacks/sheets/sheet_types.js [hunk 2]
```diff
@@ -38,7 +38,8 @@ MetalSheet.template = {
 				recipes: [
 					{name: "closet", template_name: "closet", cost: 2, time: 1500, cant_cross: ["Closet"], on_floor: true},
 					{name: "floor tiles", template_name: "plasteel_tile", cost: 1, res_amount: 4, time: 2000},
-					{name: "metal rods", template_name: "stack_rods", cost: 1, res_amount: 2, time: 6000}
+					{name: "metal rods", template_name: "stack_rods", cost: 1, res_amount: 2, time: 6000},
+					{name: "wall girders", template_name: "girder", cost: 2, time: 4000, cant_cross: ["Girder"], on_floor: true}
```
Comment (monster860) [inline code/game/objects/items/stacks/sheets/sheet_types.js]: So metal rods take longer than wall girders now? *w e w l a d*

File: code/game/objects/structures/girders.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		}
+		return prev();
+	}
+
+	deconstruct(prev) {
+		if(!this.a.loc)
+			return;
+		if(!this.a.c.Destructible.no_deconstruct) {
+			var sheets = new Atom(this.a.server, "metal_sheet");
+			sheets.c.Stack.amount = "2";
```
Comment (monster860) [inline code/game/objects/structures/girders.js]: dont do the quotes

File: code/game/turfs.js [hunk 4]
```diff
@@ -1,9 +1,70 @@
...
+
+	deconstruct(prev) {
+		if(!this.a.loc)
+			return;
+		if(!this.a.c.Destructible.no_deconstruct) {
+			var girder = new Atom(this.a.server, "girder");
+			girder.loc = this.a.loc
+
+			var sheets = new Atom(this.a.server, "metal_sheet");
+			sheets.c.Stack.amount = "2";
```
Comment (monster860) [inline code/game/turfs.js]: dont do the quotes

File: code/game/objects/structures/girders.js (hunk 3 above)
Comment (ike709) [reply code/game/objects/structures/girders.js]: woops

File: code/game/turfs.js (hunk 4 above)
Comment (ike709) [reply code/game/turfs.js]: woops x2

File: code/game/objects/items/stacks/sheets/sheet_types.js (hunk 2 above)
Comment (ike709) [reply code/game/objects/items/stacks/sheets/sheet_types.js]: you a chumbis

**Pull request merged**